### PR TITLE
Update sorts/quick_sort_3_partition.py

### PR DESCRIPTION
**Pull request by taseikyo**
Another quick sort algorithm, returns a new sorted list

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by taseikyo** [comment]
> I am not a fan of this change. What about `three_way_quicksort()` ?
> 
> Are we trying to do https://en.wikipedia.org/wiki/Quicksort#Three-way_radix_quicksort or https://en.wikipedia.org/wiki/Multi-key_quicksort ?
> Or is there some other webpage that describes the algorithm that we are implementing?

Yes, we are implementing https://en.wikipedia.org/wiki/Quicksort#Three-way_radix_quicksort. First divide the list into three parts. Then recursively sort the "less than" and "greater than" partitions.

File: sorts/quick_sort_3_partition.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-def quick_sort_3partition(sorting: list, left: int, right: int) -> list:
+def quick_sort_3partition(sorting: list, left: int, right: int):
```
Comment (cclauss) [inline sorts/quick_sort_3_partition.py:1]: Why this change?

Comment (taseikyo) [reply sorts/quick_sort_3_partition.py:1]: `quick_sort_3partition` does not return a new list, it just updates in place. So I remove `-> list`.

File: sorts/quick_sort_3_partition.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-def quick_sort_3partition(sorting, left, right):
+def quick_sort_3partition(sorting: list, left: int, right: int):
```
Comment (cclauss) [inline sorts/quick_sort_3_partition.py:1]:
```suggestion
def quick_sort_3partition(sorting: list, left: int, right: int) -> None:
```

**Review by cclauss — APPROVED** [review]
